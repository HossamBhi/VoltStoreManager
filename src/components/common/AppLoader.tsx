import React from 'react';
import {
  ActivityIndicator,
  Platform,
  ViewStyle,
  StyleProp,
  StyleSheet,
  View,
} from 'react-native';
import {useTheme} from 'react-native-paper';
import CustomeText from './CustomeText';

export default ({
  style,
  text,
}: {
  style?: StyleProp<ViewStyle>;
  text?: string;
}) => {
  const {colors} = useTheme();
  return (
    <View style={[styles.container, style]}>
      <ActivityIndicator
        size={Platform.OS === 'ios' ? 'large' : 50}
        color={colors.primary}
        animating
      />
      {text && <CustomeText text={text} />}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent',
  },
});
