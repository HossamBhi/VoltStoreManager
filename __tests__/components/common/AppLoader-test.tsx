/* eslint-disable jest/no-disabled-tests */
import React from 'react';
import {create, act} from 'react-test-renderer';
import AppLoader from '../../../src/components/common/AppLoader';

it.skip('App loader component', () => {
  // render the component
  let root = create(<AppLoader />);
  // make assertions on root
  expect(root.toJSON()).toMatchSnapshot();

  // update with some different props
  act(() => {
    root.update(<AppLoader text="Hello" />);
  });

  // make assertions on root
  expect(root.toJSON()).toMatchSnapshot();
  // const loader = create(<AppLoader />);
  // expect(loader).toMatchSnapshot();
});
