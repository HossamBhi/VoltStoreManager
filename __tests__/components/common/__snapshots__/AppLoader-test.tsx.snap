// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App loader component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="rgba(103, 80, 164, 1)"
    size="large"
  />
</View>
`;

exports[`App loader component 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="rgba(103, 80, 164, 1)"
    size="large"
  />
  <Text
    style={
      Array [
        Object {
          "color": undefined,
        },
        undefined,
      ]
    }
  >
    Hello
  </Text>
</View>
`;
